pkg/skaffold/deploy/status_check.go

```diff
@@ -257,3 +260,16 @@ func (c *resourceCounter) markProcessed(err error) resourceCounter {
 		pods:        &podCp,
 	}
 }
+
+func statusCheckDeadline(value int, deployments []Resource) time.Duration {
+	if value > 0 {
+		return time.Duration(value) * time.Second
+	}
+	d := time.Duration(0)
+	for _, r := range deployments {
+		if r.Deadline() > d {
+			d = r.Deadline()
+		}
+	}
+	return d
```

Comment:
`return time.Duration(0)`

Reply:
no. this would return the max duration seen so far.